homeassistant/components/spc/__init__.py

```diff
@@ -62,19 +79,27 @@ async def async_upate_callback(spc_object):
         _LOGGER.error("Failed to load area/zone information from SPC")
         return False
 
-    # add sensor devices for each zone (typically motion/fire/door sensors)
-    hass.async_create_task(
-        discovery.async_load_platform(hass, Platform.BINARY_SENSOR, DOMAIN, {}, config)
+    device_registry = dr.async_get(hass)
+    device_registry.async_get_or_create(
```

Comment:
Should this perhaps not be created as a device from the `alarm_control_panel`?

Reply:
I'm not sure, I haven't seen that done in other places, it all seems to be part of __init__ or a coordinator:

```
$ rg "async_get_or_create" | sort | uniq | sed -e '/test/d'
agent_dvr/__init__.py:    device_registry.async_get_or_create(
apple_tv/__init__.py:        device_registry.async_get_or_create(
auth/__init__.py:        user = await hass.auth.async_get_or_create_user(credential)
axis/device.py:        device_registry.async_get_or_create(
bluetooth/__init__.py:    dr.async_get(hass).async_get_or_create(
bond/__init__.py:    device_registry.async_get_or_create(
bosch_shc/__init__.py:    device_registry.async_get_or_create(
bthome/config_flow.py:                    return self._async_get_or_create_entry(bindkey)
bthome/config_flow.py:            return self._async_get_or_create_entry()
bthome/config_flow.py:    def _async_get_or_create_entry(self, bindkey=None):
config/auth_provider_homeassistant.py:    credentials = await provider.async_get_or_create_credentials(
control4/__init__.py:    device_registry.async_get_or_create(
deconz/deconz_event.py:        entry = device_registry.async_get_or_create(
deconz/gateway.py:        device_registry.async_get_or_create(
dlna_dmr/media_player.py:        device_entry = dev_reg.async_get_or_create(
dlna_dmr/media_player.py:        ent_reg.async_get_or_create(
eight_sleep/__init__.py:        dev_reg.async_get_or_create(
eight_sleep/__init__.py:    dev_reg.async_get_or_create(
esphome/__init__.py:    device_entry = device_registry.async_get_or_create(
fibaro/__init__.py:    device_registry.async_get_or_create(
firmata/__init__.py:    device_registry.async_get_or_create(
fronius/__init__.py:        device_registry.async_get_or_create(
google_assistant/__init__.py:    device_registry.async_get_or_create(
guardian/__init__.py:        device = dev_reg.async_get_or_create(
hassio/__init__.py:        dev_reg.async_get_or_create(config_entry_id=entry_id, **params)
hassio/__init__.py:    dev_reg.async_get_or_create(config_entry_id=entry_id, **params)
home_connect/__init__.py:            device_entry = device_registry.async_get_or_create(
homekit/__init__.py:        dev_reg.async_get_or_create(
homekit_controller/connection.py:            device = device_registry.async_get_or_create(
homekit_controller/device_trigger.py:        trigger = async_get_or_create_trigger_source(conn.hass, device_id)
homekit_controller/device_trigger.py:    return async_get_or_create_trigger_source(hass, device_id).async_attach_trigger(
homekit_controller/device_trigger.py:def async_get_or_create_trigger_source(
homematicip_cloud/__init__.py:    device_registry.async_get_or_create(
huawei_lte/__init__.py:        device_registry.async_get_or_create(
hue/__init__.py:        device_registry.async_get_or_create(
hue/v1/hue_event.py:        entry = device_registry.async_get_or_create(
hue/v2/device.py:        return dev_reg.async_get_or_create(config_entry_id=entry.entry_id, **params)
insteon/__init__.py:    device_registry.async_get_or_create(
isy994/__init__.py:    _async_get_or_create_isy_device_in_registry(hass, entry, isy)
isy994/__init__.py:    device_registry.async_get_or_create(
isy994/__init__.py:def _async_get_or_create_isy_device_in_registry(
konnected/panel.py:        device_registry.async_get_or_create(
lcn/helpers.py:        device_registry.async_get_or_create(
lcn/helpers.py:    device_registry.async_get_or_create(
livisi/__init__.py:    device_registry.async_get_or_create(
lutron_caseta/__init__.py:            dr_device = device_registry.async_get_or_create(
lutron_caseta/__init__.py:    device_registry.async_get_or_create(**device_args, config_entry_id=config_entry_id)
matter/adapter.py:        dr.async_get(self.hass).async_get_or_create(
mikrotik/__init__.py:    device_registry.async_get_or_create(
mobile_app/__init__.py:    device = device_registry.async_get_or_create(
mobile_app/config_flow.py:        devt_entry = entity_registry.async_get_or_create(
mobile_app/webhook.py:    device_registry.async_get_or_create(
motion_blinds/__init__.py:        device_registry.async_get_or_create(
motioneye/__init__.py:    device = device_registry.async_get_or_create(
mqtt/mixins.py:            device_registry.async_get_or_create(
mqtt/mixins.py:        device = device_registry.async_get_or_create(**update_device_info)
nanoleaf/__init__.py:        device_entry = device_registry.async_get_or_create(
netgear/__init__.py:    device_registry.async_get_or_create(
nibe_heatpump/__init__.py:    device_entry = reg.async_get_or_create(
nightscout/__init__.py:    device_registry.async_get_or_create(
nobo_hub/__init__.py:    dev_reg.async_get_or_create(
notion/__init__.py:    device_registry.async_get_or_create(
nuki/__init__.py:    dev_reg.async_get_or_create(
nut/__init__.py:    device_registry.async_get_or_create(
onboarding/views.py:            credentials = await provider.async_get_or_create_credentials(
onewire/onewirehub.py:            device_registry.async_get_or_create(
opentherm_gw/__init__.py:        gw_dev = dev_reg.async_get_or_create(
overkiz/__init__.py:        device_registry.async_get_or_create(
plugwise/gateway.py:    device_registry.async_get_or_create(
ps4/__init__.py:                registry.async_get_or_create(
rachio/__init__.py:        webhook_url = await async_get_or_create_registered_webhook_id_and_url(
rachio/__init__.py:    async_get_or_create_registered_webhook_id_and_url,
rachio/webhooks.py:async def async_get_or_create_registered_webhook_id_and_url(
renault/renault_hub.py:        device_registry.async_get_or_create(
roon/__init__.py:    device_registry.async_get_or_create(
ruckus_unleashed/__init__.py:        registry.async_get_or_create(
sfr_box/__init__.py:    device_registry.async_get_or_create(
shelly/coordinator.py:        device_entry = dev_reg.async_get_or_create(
sia/hub.py:            device_registry.async_get_or_create(
simplisafe/__init__.py:    base_station = device_registry.async_get_or_create(
smarttub/controller.py:            device_registry.async_get_or_create(
sonos/switch.py:        new_device = device_registry.async_get_or_create(
spc/__init__.py:    device_registry.async_get_or_create(
starline/__init__.py:        device_registry.async_get_or_create(
switcher_kis/__init__.py:        dev_reg.async_get_or_create(
syncthru/__init__.py:    device_registry.async_get_or_create(
tasmota/__init__.py:    device_registry.async_get_or_create(
tellduslive/__init__.py:        dev_reg.async_get_or_create(
toon/__init__.py:    device_registry.async_get_or_create(
tradfri/__init__.py:    dev_reg.async_get_or_create(
tuya/__init__.py:        device_registry.async_get_or_create(
unifi/controller.py:        device_registry.async_get_or_create(
upnp/__init__.py:        device_entry = dev_registry.async_get_or_create(
utility_meter/__init__.py:        entity_entry = entity_registry.async_get_or_create(
wemo/wemo_device.py:    entry = device_registry.async_get_or_create(
xiaomi_aqara/__init__.py:    device_registry.async_get_or_create(
xiaomi_ble/__init__.py:            device = device_registry.async_get_or_create(
xiaomi_ble/config_flow.py:                    return self._async_get_or_create_entry(bindkey)
xiaomi_ble/config_flow.py:            return self._async_get_or_create_entry()
xiaomi_ble/config_flow.py:    def _async_get_or_create_entry(self, bindkey=None):
xiaomi_miio/__init__.py:    device_registry.async_get_or_create(
zha/__init__.py:    device_registry.async_get_or_create(
zha/core/gateway.py:            device_registry_device = self.ha_device_registry.async_get_or_create(
zha/core/gateway.py:            zha_device = self._async_get_or_create_device(zigpy_device, restored=True)
zha/core/gateway.py:            zha_group = self._async_get_or_create_group(group)
zha/core/gateway.py:        zha_device = self._async_get_or_create_device(device)
zha/core/gateway.py:        zha_group = self._async_get_or_create_group(zigpy_group)
zha/core/gateway.py:    def _async_get_or_create_device(
zha/core/gateway.py:    def _async_get_or_create_group(self, zigpy_group: zigpy.group.Group) -> ZHAGroup:
zwave_js/__init__.py:        device = self.dev_reg.async_get_or_create(
```

Reply:
I guess it depends if you intend to have it as a device for a hub or if it's a device representing the alarm panel.
Probably the first one which should be ok then but wanted to ask.

Reply:
OK. Resolving.